src/main/flight/gps_rescue.c

```diff
@@ -537,169 +649,197 @@ static bool checkGPSRescueIsAvailable(void)
     return result;
 }
 
-/*
-    Determine what phase we are in, determine if all criteria are met to move to the next phase
-*/
 void updateGPSRescueState(void)
+// this runs a lot faster than the GPS Data update rate, and runs whether or not rescue is active
 {
-    static uint16_t newDescentDistanceM;
-    static float_t lineSlope;
-    static float_t lineOffsetM;
-    static int32_t newSpeed;
-    static int32_t newAltitude;
-    float magnitudeTrigger;
-
     if (!FLIGHT_MODE(GPS_RESCUE_MODE)) {
-        rescueStop();
+        rescueStop(); // sets phase to idle and does nothing else
     } else if (FLIGHT_MODE(GPS_RESCUE_MODE) && rescueState.phase == RESCUE_IDLE) {
-        rescueStart();
-        rescueAttainPosition(); // Initialize
-        performSanityChecks(); // Initialize
+        rescueStart(); // sets phase to initialise if we enter GPS Rescue mode while idle
+        rescueAttainPosition(); // Initialise basic parameters when a Rescue starts (can't initialise sensor data reliably)
+        performSanityChecks(); // Initialises sanity check values when a Rescue starts
     }
 
-    rescueState.isFailsafe = failsafeIsActive();
+    // Will now be in INITIALIZE mode, if just entered Rescue while IDLE, otherwise stays IDLE
 
-    sensorUpdate();
+    sensorUpdate(); // always get latest GPS / Altitude data
 
+    uint8_t halfAngle = gpsRescueConfig()->angle / 2;
```

Comment:
`angle` is type of `uint16`. So it can overflow or not?

Reply:
In theory yes, but practically it ~is not able to~ should not be able to overflow because `UINT8_MIN <= halfAngle <= 180 < UINT8_MAX`